scipy/stats/qmc.py

```diff
@@ -0,0 +1,1368 @@
+        max values of the sample will coincide with the bounds.
+    reverse : bool
+        Reverse the transformation, from bounds range to unit hypercube.
+
+    Returns
+    -------
+    sample : array_like (n_samples, dim)
+        Scaled-sample.
+
+    Examples
+    --------
+
+    >>> from scipy.stats import qmc
+    >>> bounds = [[-2, 0],
+    ...           [6, 5]]
+    >>> sample = [[0.5 , 0.5 ],
+    ...           [0.75, 0.25]]
+    >>> qmc.scale(sample, bounds)
+    array([[2.  , 2.5 ],
+           [4.  , 1.25]])
+
+    """
+    bounds = np.asarray(bounds)
+    min_ = np.min(bounds, axis=0)
+    max_ = np.max(bounds, axis=0)
+    if not reverse:
+        return sample * (max_ - min_) + min_
+    else:
+        return (sample - min_) / (max_ - min_)
+
+
+def discrepancy(sample, iterative=False, method='CD'):
+    """Discrepancy on a given sample.
+
+    Parameters
+    ----------
+    sample : array_like (n_samples, dim)
+        The sample to compute the discrepancy from.
+    iterative : bool
+        Must be False if not using it for updating the discrepancy.
+    method : str
+        Type of discrepancy, can be ['CD', 'WD', 'MD', 'star'].
+
+    Returns
+    -------
+    discrepancy : float
+        Centered discrepancy.
+
+    Notes
+    -----
+
+    The discrepancy is a uniformity criterion used to assess the space filling
+    of a number of samples in a hypercube.
+    The discrepancy measures how the spread of the points deviates from a
+    uniform distribution.
+    The lower the value is, the better the coverage of the parameter space is.
+
+    Taking a subspace of the parameter space we count the number of points in
+    the subpace and compare it to the total number of the points of the sample.
+    This value is substracted by the volume of the subspace. This process is
+    done over all subspaces and the highest value is kept.
+
+    Three methods are available:
+
+    * CD: Centered Discrepancy - subspace involves a corner of the hypercube
+    * WD: Wrap-around Discrepancy - subspace can wrap around bounds
+    * MD: Mixture Discrepancy - mix between CD/WD covering more criteria
+    * star: Star L2-discrepancy - like CD BUT variant to rotation
+
+    References
+    ----------
+    .. [1] Fang et al. Design and modeling for computer experiments,
+       Computer Science and Data Analysis Series, 2006.
+    .. [2] Zhou Y.-D. et al. Mixture discrepancy for quasi-random point sets
+       Journal of Complexity, 29 (3-4) , pp. 283-301, 2013.
+    .. [3] T. T. Warnock. Computational investigations of low discrepancy point
+       sets, S.K. Zaremba (editor), Applications of Number Theory to Numerical
+       Analysis, Academic Press, New York.
+
+    Examples
+    --------
+    Calculate the quality of the sample using the discrepancy:
+
+    >>> from scipy.stats import qmc
+    >>> space = np.array([[1, 3], [2, 6], [3, 2], [4, 5], [5, 1], [6, 4]])
+    >>> space = (2.0 * space - 1.0) / (2.0 * 6.0)
```

Comment:
This is unclear, should be written as `scale(...)`?

Reply:
This was to have something cleaner than `bounds = np.array([[0.5, 0.5], [6.5, 6.5]])`. I can change this to use scale instead to be consistent.